CDCgov/prime-reportstream
# Experience/6392 fetch fn provider

In order to move away from both rest-hooks and our own implementations of fetch logic towards react-query, we want all components and hooks in the application to maintain a simple interface for making api requests.

A simple interface like this would not include having to in any way think about globally available data on every request.

In service of that idea, this PR introduces a context based hook that will provide hooks (and technically components, but the idea is that this would generally just be used in hooks [digression]) with an authorized fetch function to use to make all API requests.

As a POC this is introduced on the value sets pages.

As you'll see from some of the code comments, some of this is open to change in the next PR when we will introduce the idea of resources, but this is generally how I think things will look.

Also note that some of the later commits here are cleaning up code based on API changes downstream, particularly relating to the value sets row paths that the API expects now.

## TODO


## Test Steps:
value sets regression testing. see https://github.com/CDCgov/prime-reportstream/pull/6476

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
